dbms/src/Storages/MergeTree/BackgroundProcessingPool.cpp

```diff
@@ -61,9 +61,12 @@ void BackgroundProcessingPoolTaskInfo::wake()
 }
 
 
-BackgroundProcessingPool::BackgroundProcessingPool(int size_) : size(size_)
+BackgroundProcessingPool::BackgroundProcessingPool(int size_, const char * thread_name_)
+    : size(size_)
+    , thread_name(thread_name_)
 {
-    LOG_INFO(&Logger::get("BackgroundProcessingPool"), "Create BackgroundProcessingPool with " << size << " threads");
+    logger = &Logger::get(String("BackgroundProcessingPool[") + thread_name + "]");
```

Comment:
not sure about braces, may be bit safer to use alphanumerical (log parsers, thread name in /proc/ etc.)